src/components/PostRelatedArticles/index.jsx

```diff
@@ -0,0 +1,71 @@
+import { graphql, useStaticQuery } from "gatsby"
+import React from "react"
+
+import * as styles from "./index.module.scss"
+
+export const query = graphql`
+  {
+    allMarkdownRemark(
+      filter: { fileAbsolutePath: { regex: "/src.posts/" } }
+      sort: [{ frontmatter: { date: DESC } }, { frontmatter: { title: DESC } }]
+    ) {
+      nodes {
+        fields {
+          path
+        }
+        frontmatter {
+          categories {
+            key
+            label
+          }
+          title
+        }
+      }
+    }
+  }
+`
+
+function Article({ article }) {
+  return (
+    <div className={styles.article} key={article.fields.path}>
+      <p className={styles.articleTitle}>{article.frontmatter.title}</p>
+      <p className={styles.articleLink}>
+        <a href={article.fields.path}>Read more</a>
+      </p>
+    </div>
+  )
+}
+
+function PostRelatedArticles({ currentArticleTitle }) {
+  const data = useStaticQuery(query)
+
+  const currentArticleCategories = data.allMarkdownRemark.nodes
+    .filter((article) => article.frontmatter.title === currentArticleTitle)
+    .map((article) =>
+      article.frontmatter.categories?.map((category) => category.label)
+    )[0]
+
+  const relatedArticles = data.allMarkdownRemark.nodes
+    .filter((article) => article.frontmatter.title !== currentArticleTitle)
+    .filter((article) =>
+      article.frontmatter.categories
+        ?.map((category) => category.label)
+        .some((label) => currentArticleCategories?.includes(label))
+    )
+    .slice(0, 3)
+
+  return (
+    <div className={styles.root}>
+      {relatedArticles.length > 0 && (
+        <p className={styles.title}>Related Articles</p>
+      )}
```

Comment:
Why is the rest of the element returned if there are no related articles?

Reply:
And what about the parent styles?

```
<div className={styles.relatedArticles}>
    <PostRelatedArticles currentArticleTitle={title} />
</div>
 ```

Reply:
This may raise a "higher level" question ... what happens if there are no related articles? What do we display?

Reply:
If there are no articles they just don't render, but I agree I should move this check one level up. 

There's indeed the problem with the parent styles that I haven't been able to fix yet - when there are no articles I still have the spacing as if there were... will look into it

Reply:
Normally, this would be a question for Onda ... one thing we could do is display 3 random articles instead .. and instead of `Related`, it would be something like `More` - But I don't know if this is something we want. @gabrielpoca

Reply:
About the problem with the parent styles, if you have a layout element, that sets the spacing on `gap`, then you won't have this problem. (But let's pair today)